tests/test_resolvers.py

```diff
@@ -0,0 +1,587 @@
+        self.test_private_dir = os.path.join(self.resources_path, 'html/private')
+        self.dir_paths: DirPaths = {
+            'cache': '',
+            'json': '',
+            'log': yc_gc.logs_dir,
+            'private': '',
+            'result': yc_gc.result_dir,
+            'save': yc_gc.save_file_dir,
+            'yang_models': yc_gc.yang_models,
+        }
+        self.test_repo = os.path.join(yc_gc.temp_dir, 'test/YangModels/yang')
+
+    def setUp(self) -> None:
+        self.logger = logging.getLogger('test')
+
+        return super().setUp()
+
+    # BasicResolver
+    def test_basic_resolver_simple_resolve(self):
+        # represents 'module test-module: output test-output'
+        module_stmt = new_statement(None, None, None, 'module', 'test-module')
+        output_stmt = new_statement(None, module_stmt, None, 'output', 'test-output')
+        module_stmt.substmts.append(output_stmt)
+
+        br = BasicResolver(parsed_yang=module_stmt, property_name='output')
+        res = br.resolve()
+        self.assertEqual(res, 'test-output')
+
+    def test_basic_resolver_simple_resolve_with_underscore(self):
+        # consider 'leaf-list' if this keyword does not work
+        module_stmt = new_statement(None, None, None, 'module', 'test-module')
+        output_stmt = new_statement(None, module_stmt, None, 'output-test', 'test-output')
+        module_stmt.substmts.append(output_stmt)
+
+        br = BasicResolver(parsed_yang=module_stmt, property_name='output_test')
+        res = br.resolve()
+        self.assertEqual(res, 'test-output')
+
+    # GeneratedFromResolver
+    def test_generated_from_resolver_simple_resolve_default(self):
+        gfr = GeneratedFromResolver(self.logger, name='test_default', namespace='some_namespace')
+        res = gfr.resolve()
+        self.assertEqual(res, 'not-applicable')
+
+    def test_generated_from_resolver_simple_resolve_default_no_namespace(self):
+        gfr = GeneratedFromResolver(self.logger, name='test_default', namespace=None)
+        res = gfr.resolve()
+        self.assertEqual(res, 'not-applicable')
+
+    def test_generated_from_resolver_simple_resolve_mib(self):
+        gfr = GeneratedFromResolver(self.logger, name='test_mib', namespace='something:smi')
+        res = gfr.resolve()
+        self.assertEqual(res, 'mib')
+
+    def test_generated_from_resolver_simple_resolve_native(self):
+        gfr = GeneratedFromResolver(self.logger, name='test_native_cisco', namespace='something:cisco:something')
+        res = gfr.resolve()
+        self.assertEqual(res, 'native')
+
+    def test_generated_from_resolver_simple_resolve_native_caps(self):
+        gfr = GeneratedFromResolver(self.logger, name='CISCOtest_native', namespace='Something:CISCO:sOmEtHiNg')
+        res = gfr.resolve()
+        self.assertEqual(res, 'native')
+
+    # ImplementationResolver
+    def test_implementation_resolver_simple_resolve(self):
+        platform_data1 = {
```

Comment:
Also something more real for `implementation` for example:
```
  {
      "conformance-type": "implement",
      "feature-set": "ALL",
      "os-type": "NX-OS",
      "os-version": "10.3(1)",
      "platform": "Nexus 9000",
      "software-flavor": "ALL",
      "software-version": "10.3(1)",
      "vendor": "cisco"
  }
```
You can find modules on [PROD](https://yangcatalog.org/yang-search/module_details)